### DISCUSSION TITLE
How to address multiple instances of a value within RAMSECTION?

### DISCUSSION THREAD

**Discussion by Pokeman2003** [vhelin/wla-dx#616] [Q&A] [answered]
Hypothetically, let's say I have a RAM section that looks like this:
```asm
.RAMSECTION "Test RAMSection"
structList INSTANCEOF Struct 32
.ENDS
```
I understand how to access instance number 1, but how would I access the rest of the instances?

**Comment by vhelin** [comment, marked as answer]
There's an easy to to find out if the documentation here is lacking (I need to check if it is):

![kuva](https://github.com/vhelin/wla-dx/assets/6417746/0893f9e4-6310-4e8d-9d24-f090701ad6bc)

... then write out a symbol file and take a look:

![kuva](https://github.com/vhelin/wla-dx/assets/6417746/9a9aed94-ffee-4fd0-81e7-cb190466b085)

**Reply by Pokeman2003** [reply]
Oh I'm an idiot. I thought it started at 0. No wonder my code kept complaining about an unknown symbol.

Update: Wait, no. That doesn't work for my struct. Granted, I'm using DSB in my struct, which I assumed could be addressed the same way. Is that the case or am I just a goof?
Update 2: After some experimentation, turns out that's just me being a goof. You can only index STRUCTs, not data. Derp.

**Reply by Pokeman2003** [reply]
Upon further inspection, it appears you do actually address it in the form of this off-hand remark in the comments. `; If you leave away "STARTFROM 0" the indexing will start from 1` shows up in the RAMSection's second code example, which is something I totally missed last night.

However, when I'm searching for information about structs, I'm only going to be skim-reading other sections. For this reason, it would probably behoove you to put this information in STRUCT, which explains in more detail on how to put this in a RAMsection. Something akin to `<instance name> INSTANCEOF <struct name> [optional, <the number of structures>] [optional, STARTFROM <the starting index number>]` would work, on top of explaining how indexing looks syntactically(one of my early attempts involved doing something like `structList[10]`, which obviously wouldn't work.)
